### PR TITLE
Teach OpenApiJsonWriter to produce a terse output

### LINKED ISSUE #626: How about teaching the Json Writer to generate dense/terse serialized output?
:wave:

The current output is in "pretty" mode (with indentations, newlines, ....).
Our need was to generate a compact representation of it.

I've been able to achieve this through the following code:

```csharp
public class CustomStringWriter : StringWriter
  {
    public CustomStringWriter(IFormatProvider formatProvider) : base(formatProvider)
    {
    }

    public override void WriteLine()
    {
      return;
    }

    public override void Write(string value)
    {
      // Duh :/
      // /!\ Danger => **Very** fragile hack! as it depends in an implementation detail in OpenAPI.NET

      if (value == ": ")
      {
        base.Write(":");
        return;
      }

      base.Write(value);
    }
  }

  public class CustomOpenApiJsonWriter : OpenApiJsonWriter
  {
    public CustomOpenApiJsonWriter(TextWriter writer) : base(writer)
    {

    }

    public override void WriteIndentation()
    {
      return;
    }
  }
```

However, in order to achieve maximum "compactness", I had to rely on a very crude hack (see above in the code).

Because of

https://github.com/microsoft/OpenAPI.NET/blob/649f0b1780c575c4780f4091eedd7a555e8ccc4a/src/Microsoft.OpenApi/Writers/OpenApiJsonWriter.cs#L156

and 

https://github.com/microsoft/OpenAPI.NET/blob/649f0b1780c575c4780f4091eedd7a555e8ccc4a/src/Microsoft.OpenApi/Writers/WriterConstants.cs#L84

Although this code works, it may not resist a refactoring of OpenAPI.NET code.

Would you be open to teach the json writers to generate a dense json output?
Would that the case, I'd be happy to work on a PR.

### PR DESCRIPTION
**Pull request by nulltoken**
Fix #626

### CODE REVIEW

File: src/Microsoft.OpenApi/Microsoft.OpenApi.csproj [hunk 1]
```diff
@@ -37,6 +37,7 @@
 
     <ItemGroup>
         <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All"/>
+        <PackageReference Include="System.ValueTuple" Version="4.5.0" />
```
[the remark this answers is not in the document]
Comment (darrelmiller) [reply src/Microsoft.OpenApi/Microsoft.OpenApi.csproj]: I'm perfectly fine with bumping the test project to ,net48

**Comment by nulltoken** [comment]
@darrelmiller I think I'm done. This is now ready for review. And it ended up with a lot of modified files.

In order to make the review easier (or at least as easier as possible :-/), I've segregated the tests changes in separate commits.

Hope that'll help.

File: test/Microsoft.OpenApi.SmokeTests/Microsoft.OpenApi.SmokeTests.csproj [hunk 2]
```diff
@@ -1,11 +1,11 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net462</TargetFrameworks>
+    <TargetFrameworks>net48</TargetFrameworks>
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
-    <DefineConstants>TRACE;DEBUG;net462</DefineConstants>
+    <DefineConstants>TRACE;DEBUG;NET48</DefineConstants>
```
Comment (SimonCropp) [inline test/Microsoft.OpenApi.SmokeTests/Microsoft.OpenApi.SmokeTests.csproj:8]: i know the DefineConstants was already defined. but dont these get added by default? Want me to submit a PR (after this) that removes these?

**Comment by SimonCropp** [comment]
Would [VerifyJson](https://github.com/VerifyTests/Verify#verifyjson) be useful in this PR?

**Comment by nulltoken** [comment]
> Would [VerifyJson](https://github.com/VerifyTests/Verify#verifyjson) be useful in this PR?

I've pondered about using it. However, as the initial code was comparing string to string, I opted for the basic `Verify`. Besides, in the context on this PR and the tests surrounding the proper serialization into a dense/terse json, I don't think we would want 

```
{
  "msg": "hello"
}
```

to be seen as equal to 

```
{"msg":"hello"}
```

**Comment by SimonCropp** [comment]
@nulltoken re  VerifyJson. yes you are correct

**Comment by darrelmiller** [comment]
I am going to test this an merge it before any other PRs so that we don't have any more conflicts.

**Comment by darrelmiller** [comment]
@nulltoken Thank you for this work.  The pretty printed Graph V1 OpenAPI was 25MB and the terse one was 14MB!  And thanks @SimonCropp I had not seen Verify before.  It took me a little while staring at this PR to understand the concept but I think this type of testing is idea for this type of library.

**Pull request merged**